### ISSUE TITLE
[Feature] Add output_padding option to ConvTranspose2d

### ISSUE THREAD

**Issue by zcbenz** [ml-explore/mlx#1423]
Passing the same parameters to `nn.ConvTranspose2d` does not revert `nn.Conv2d` in terms of dimension shape conservation, and an `output_padding` option can solve it.

Reference from [PyTorch's docs](https://pytorch.org/docs/stable/generated/torch.nn.ConvTranspose2d.html):

> The padding argument effectively adds dilation * (kernel_size - 1) - padding amount of zero padding to both sizes of the input. This is set so that when a [Conv2d](https://pytorch.org/docs/stable/generated/torch.nn.Conv2d.html#torch.nn.Conv2d) and a [ConvTranspose2d](https://pytorch.org/docs/stable/generated/torch.nn.ConvTranspose2d.html#torch.nn.ConvTranspose2d) are initialized with same parameters, they are inverses of each other in regard to the input and output shapes. However, when stride > 1, [Conv2d](https://pytorch.org/docs/stable/generated/torch.nn.Conv2d.html#torch.nn.Conv2d) maps multiple input shapes to the same output shape. output_padding is provided to resolve this ambiguity by effectively increasing the calculated output shape on one side. Note that output_padding is only used to find output shape, but does not actually add zero-padding to output.

**Comment by ParamThakkar123** [comment]
I would like to solve this issue

**Comment by ParamThakkar123** [comment]
Do we need to make these changes to conv_transpose_general and then to conv_transpose_2d or separately to conv_transpose_2d without making any change to conv_transpose_general ??

**Comment by awni** [comment]
This should be an argument for `conv_transpose_{1,2,3}d`. I would have the core logic in `conv_transpose_general` but that is an implementation detail.
